src/App.tsx

```diff
@@ -14,11 +19,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  getSelected = (good: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
```

Comment:
тут краще назву змінювати для параметру 
```suggestion
    this.setState((currentState) => ({
      selectedGoods: [...currentState.selectedGoods, good],
    }));
```